Icom R8600 to Alpha

malcolmh opened the pull request:
All standard functions have been configured and tested.

DF4OR commented:
Hi Malcolm.
Any good reason why this patch deletes all my work in progress I did on this backend?

73,
Ekki, DF4OR

N0NB commented:
* On 2019 10 Jan 11:23 -0600, Ekki Plicht (DF4OR) wrote:
> Hi Malcolm.
> Any good reason why this patch deletes all my work in progress I did on this
> backend?

I will not apply this patch until both agree on what must be done to resolve this.

73, Nate

- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: http://www.n0nb.us  GPG key: D55A8819  GitHub: N0NB

DF4OR commented:
Nate, 
I retract my objection. Please go ahead and merge this PR. Life is too short to bother with stuff like this. I'm outta here, spending my time with other software and other hobbies.

73,
Ekki, DF4OR

N0NB commented:
* On 2019 13 Jan 12:49 -0600, Ekki Plicht (DF4OR) wrote:
> Nate, 
> I retract my objection. Please go ahead and merge this PR. Life is too
> short to bother with stuff like this. I'm outta here, spending my time
> with other software and other hobbies.

I'm sorry to see you leave, Ekki.  Thanks for your contributions and
we'll still be here should you wish to contribute again.

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: http://www.n0nb.us  GPG key: D55A8819  GitHub: N0NB

N0NB commented:
* On 2019 10 Jan 09:39 -0600, Malcolm Herring wrote:
> All standard functions have been configured and tested.
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/Hamlib/Hamlib/pull/84
> 

Hi Malcolm.

Please explain why much of Ekki's work is being deleted in this patch.
I would like to apply only what is needed.

Thanks,

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: http://www.n0nb.us  GPG key: D55A8819  GitHub: N0NB

malcolmh commented:
I have not deleted Ekki's work. The rig code was cloned from ic75.c and there was some parts of that which is not relevant to the R8600. That is what I deleted. All the changes & additions that Ekki made are still there & I have added to that. I also cleaned it up, re-indenting & removing out-of-date comments. One thing I should have done was add Ekki to the contributors.

mikaelnousiainen commented:
A good question is of course: did the `set_channel` and `get_channel` implementations still work for this model?

DF4OR commented:
a) The set/get_channel was partially functioning and was work in progress. I have a local copy where it works as this is my main interest - getting a program for memory management to work.

b) the deleted comments were reminders/notes for my pleasure, so I do not forget what I was working on and what I still need to explore/better understand. I didn't now that they are not wanted.

But as I said - life is too short for such unpleasantness. Let him has his way, I will no further contribute to Hamlib.

73,
Ekki, DF4OR

N0NB commented:
Deleting comments such as those, or any others that relate to how the
software is intended to work, is counter productive for a long
"product/project" support life.  (Not just my view.)

My tutor at college always used to preach "Use Comments to fully
describe the program flow and purpose, so that others may understand it
enough to translate it to other languages in the future."   (This in the
days of K&R 'C'.)

Heck, some of the simple stuff I've written for work as little as two
years ago, if I have to "go back in", oh boy, do I *Need* those comments!

They may take some space, and some think they detract from the "beauty"
of code, but they are far too valuable to loose.

Not every potential contributor lives/eats/breathes code 24/7.

YVMV.

Dave G0WBX.


On 25/01/2019 16:48, Ekki Plicht (DF4OR) wrote:
>
> a) The set/get_channel was partially functioning and was work in
> progress. I have a local copy where it works as this is my main
> interest - getting a program for memory management to work.
>
> b) the deleted comments were reminders/notes for my pleasure, so I do
> not forget what I was working on and what I still need to
> explore/better understand. I didn't now that they are not wanted.
>
> But as I said - life is too short for such unpleasantness. Let him has
> his way, I will no further contribute to Hamlib.
>
> 73,
> Ekki, DF4OR
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Hamlib/Hamlib/pull/84#issuecomment-457637273>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAh7LT948U8gZHPU7a02CTyx2D0N6OiFks5vGzVdgaJpZM4Z5vuS>.
>
>
>
> _______________________________________________
> Hamlib-developer mailing list
> Hamlib-developer@lists.sourceforge.net
> https://lists.sourceforge.net/lists/listinfo/hamlib-developer

-- 
Created on and sent from a Unix like PC running and using free and open source software:

mikaelnousiainen commented:
@DF4OR @N0NB Agreed, which is what I was trying to state earlier. In my opinion, the set of comments, which were not addressed by the latest changes by @malcolmh  should be preserved. And if the set/get channel functions work even partially, they could be preserved too with comments mentioning they are a WIP.
